### PR TITLE
Add `Related Issues/PRs` section in pull request template

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## What changes are proposed in this pull request?

Add `Related Issues/PRs` section in the pull request template to encourage contributors to link issues and PRs.

## How is this patch tested?

Manual inspection:

https://github.com/mlflow/mlflow/blob/0b8fed07b8cde03f89bbf6bb73075c97e60e8963/.github/pull_request_template.md

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by BenWilson2 — APPROVED** [review]
LGTM

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## Related Issues/PRs
+
+<!-- If the linked issue can be closed by merging this PR, use a closing keyword (e.g. close, fix, resolve) followed by the issue reference. For example, "Close #123". -->
```
Comment (harupy) [inline .github/pull_request_template.md:3]: @BenWilson2 Just wanted to make sure, this sounds ok.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## Related Issues/PRs
+
```
Comment (BenWilson2) [inline .github/pull_request_template.md:2]:
```suggestion
(Please add any linked feature requests, issues, or related PRs here with the index reference. For example: #1234)
```

**Review by BenWilson2 — APPROVED** [review]
LGTM!

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,3 +1,10 @@
...
+<!--
+Please add any linked feature requests, issues, or related PRs here with the index reference. For example: #1234. If the linked items can be closed by merging this PR, please use a closing keyword (close, fix, or resolve) followed by the index reference. For example: Close #1234.
+-->
```
Comment (harupy) [inline .github/pull_request_template.md:5]: @BenWilson2 Included your suggestion as a comment.

File: .github/pull_request_template.md (hunk 2 above)
Comment (harupy) [reply .github/pull_request_template.md:2]: Thanks for the suggestion!

**Pull request merged**